activejob/lib/active_job/core.rb

```diff
@@ -43,6 +43,16 @@ module Core
     # Track when a job was enqueued
     attr_accessor :enqueued_at
 
+    # Track whether the adapter received the job successfully.
+    attr_writer :successfully_enqueued
```

Comment:
Not sure if is relevant to make this and  private `attr_accesor` so the caller would not have access to `successfully_enqueued=` method.

Reply:
That seems reasonable, however many of the other properties defined in the same section are probably properties that are generally not expected to be updated by callers, but they are still public. Making the writer private seems to me like it would be out of place, no?